src/KeyHandler.cpp

```diff
@@ -240,7 +231,7 @@ bool KeyHandler::handle(Key key, McBopomofo::InputState* state,
       return true;
     }
 
-    if (!reading_.isEmpty()) {
+    if (!reading_.isEmpty() || reading_.hasToneMarkerOnly()) {
```

Comment:
Hmm this is strange. Are you sure that only `!reading_.isEmpty()` doesn't work after all your changes in place?

If `reading_.hasToneMarkerOnly()` is `true`, it should already imply that `reading_` is _not_ empty. If that's not the case with your gcc, could you change `BopomofoSyllable::isEmpty()` from:

`return !syllable_;`

to:

`return syllable_ == 0;`

and see if it's indeed the issue there?

Reply:
OK. I just change the line back to ` if (!reading_.isEmpty())` only.